bio/porechop_abi/wrapper.py

```
@@ -0,0 +1,43 @@
+
+extra = snakemake.params.get("extra", "")
+ab_initio = snakemake.params.get("ab_initio", False)
+log = snakemake.log_fmt_shell(stdout=True, stderr=True)
+
+ab_initio_flag = "--ab_initio" if ab_initio else ""
+ab_initio_config = snakemake.input.get("ab_initio_config", "")
+ab_initio_config_arg = (
+    f"--ab_initio_config {ab_initio_config}" if ab_initio_config else ""
+)
+
+custom_adapters = snakemake.input.get("custom_adapters", "")
```

Ruff (0.15.15)

[error] Undefined name `snakemake`

(F821)